testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeRequireQueryPartitionsFilter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.deltalake;
+
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertQueryFailure;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.deltalake.util.DeltaLakeTestUtils.dropDeltaTableWithRetry;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDeltaLakeRequireQueryPartitionsFilter
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testRequiresQueryPartitionFilter()
+    {
+        String tableName = "test_require_partition_filter_" + randomNameSuffix();
+        String tableDirectory = "databricks-compatibility-test-" + tableName;
+
+        onTrino().executeQuery(format("CREATE TABLE delta.default.%s (x INT, part INT) WITH (PARTITIONED_BY = ARRAY['part'], location = 's3://%s/%s')",
+                tableName,
+                bucketName,
+                tableDirectory));
+        try {
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (1, 11), (2, 22)");
+
+            onTrino().executeQuery("SET SESSION delta.query_partition_filter_required = true");
+
+            assertQueryFailure(() -> onTrino().executeQuery("SELECT COUNT(*) FROM " + tableName))
+                    .hasMessageMatching(format("Query failed \\(#\\w+\\): Filter required on default\\.%s for at least one partition column: part", tableName));
+            assertThat(onTrino().executeQuery(format("SELECT * FROM %s WHERE part = 11", tableName))).containsOnly(row(1, 11));
+        }
+        finally {
+            dropDeltaTableWithRetry(tableName);
```

Comment:
`dropDeltaTableWithRetry` uses Delta query executor internally. We use the same query executor between CREATE and DROP.

Reply:
I added queries running against Delta executor